test/E2E/Helpers.purs

```diff
@@ -0,0 +1,365 @@
+  handler errorRef outputType f = mkEffectFn1 $ \e -> launchAff_ do
+    output <- f e
+    liftEffect $ Ref.modify_
+      ( _ <>
+          [ E2EOutput
+              { outputType: outputType
+              , output: output
+              }
+          ]
+      )
+      errorRef
+
+checkSuccess :: RunningExample -> Aff Unit
+checkSuccess (RunningExample { main, errors }) = do
+  feedback <- testFeedbackIsTrue main
+  unless feedback $ liftEffect $ showOutput errors >>= log
+  shouldSatisfy feedback (_ == true)
+
+inWalletPage
+  :: forall (a :: Type). RunningExample -> (Toki.Page -> Aff a) -> Aff a
+inWalletPage (RunningExample { browser, jQuery }) =
+  (waitForWalletPage jQuery browser >>= _)
+
+namiConfirmAccess :: RunningExample -> Aff Unit
+namiConfirmAccess = flip inWalletPage (clickButton "Access")
+
+namiSign :: RunningExample -> Aff Unit
+namiSign = flip inWalletPage \nami -> do
+  clickButton "Sign" nami
+  reactSetValue password testPassword nami
+  clickButton "Confirm" nami
+
+geroConfirmAccess :: RunningExample -> Aff Unit
+geroConfirmAccess =
+  flip inWalletPage $ \page -> do
+    delay $ wrap 100.0
+    void $ doJQ (inputType "radio") click page
+    void $ doJQ (buttonWithText "Continue") click page
+    delay $ wrap 100.0
+    void $ doJQ (inputType "checkbox") click page
+    void $ doJQ (buttonWithText "Connect") click page
+
+geroInit :: Toki.Browser -> Aff Unit
+geroInit browser = do
+  page <- Toki.newPage browser
+  jQuery <- retrieveJQuery page
+  Toki.goto
+    (wrap "chrome-extension://iifeegfcfhlhhnilhfoeihllenamcfgc/index.html")
+    page
+  void $ Toki.unsafeEvaluateStringFunction jQuery page
+  delay $ wrap 1000.0
+  void $ doJQ (buttonWithText "Get Started") click page
+  delay $ wrap 1000.0
+  void $ doJQ' (wrap "a[href=\"#/wallet-options/import-wallet\"]") click page
+  delay $ wrap 100.0
+  reactSetValue (byId "wallet-name") "ctltest" page
+  typeInto textarea geroSeed page
+  void $ doJQ (buttonWithText "Import") click page
+  delay $ wrap 100.0
+  typeInto (byId "wallet-password") testPasswordGero page
+  typeInto (byId "wallet-password-confirm") testPasswordGero page
+  void $ doJQ (byId "user-agree") click page
+  void $ doJQ (byId "password-agree") click page
+  void $ doJQ (byId "set-up-password") click page
+  delay $ wrap 100.0
+  void $ doJQ (buttonWithText "Continue") click page
+  delay $ wrap 100.0
+  void $ doJQ (byId "done") click page
+  openPopup browser
+  delay $ wrap 50000.0
+  pure unit
+
+{-geroSetCollateral :: Toki.Browser -> Aff Unit
+geroSetCollateral = do
+  page <- Toki.newPage browser
+  jQuery <- retrieveJQuery page
+  Toki.goto (wrap "chrome-extension://iifeegfcfhlhhnilhfoeihllenamcfgc/index.html") page
+-}
+
+geroSign :: RunningExample -> Aff Unit
+geroSign (RunningExample { browser }) = do
+  pure unit
+
+-- | Wrapper for Page so it can be used in `shouldSatisfy`, which needs 'Show'
+-- | Doesn't show anything, thus 'NoShow'
+newtype NoShowPage = NoShowPage Toki.Page
```

Comment:
This comment is a bit confusing to me. By "doesn't show anything", do you mean that `Page` doesn't have a `Show` instance? Should the name be something like `ShowablePage` since it provides a `Show` instance for an otherwise un`show`able type?

Reply:
It's  a Show instance with nothing to show. But I think its obsolete anyway, I will delete it.